File: articles/institutional-responses-to-public-goods-problems/cpp/public_goods_score.cpp
#include <iostream>

struct PublicGoodsCase {
    double contribution_rate;
    double legitimacy;
    double trust;
    double coordination;
    double monitoring;
    double distributional_attention;
    double scale_complexity;
    double capture_risk;
};

double public_goods_provision_score_raw(const PublicGoodsCase& x) {
    return
        0.22 * x.contribution_rate +
        0.13 * x.legitimacy +
        0.12 * x.trust +
        0.11 * x.coordination +
        0.10 * x.monitoring +
        0.08 * x.distributional_attention -
        0.12 * x.scale_complexity -
        0.08 * x.capture_risk;
}

int main() {
    PublicGoodsCase demo {70, 78, 75, 72, 80, 74, 35, 25};
    std::cout << "Public goods provision raw score: "
              << public_goods_provision_score_raw(demo) << std::endl;
    return 0;
}

File: articles/institutional-responses-to-public-goods-problems/fortran/public_goods_score.f90
program public_goods_score
  implicit none

  real :: contribution_rate, legitimacy, trust, coordination, monitoring
  real :: distributional_attention, scale_complexity, capture_risk
  real :: score

  contribution_rate = 70.0
  legitimacy = 78.0
  trust = 75.0
  coordination = 72.0
  monitoring = 80.0
  distributional_attention = 74.0
  scale_complexity = 35.0
  capture_risk = 25.0

  score = 0.22 * contribution_rate + &
          0.13 * legitimacy + &
          0.12 * trust + &
          0.11 * coordination + &
          0.10 * monitoring + &
          0.08 * distributional_attention - &
          0.12 * scale_complexity - &
          0.08 * capture_risk

  print *, "Public goods provision raw score:", score
end program public_goods_score
